config/bot_filter.go

```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"golang.ngrok.com/ngrok/internal/pb"
+)
+
+// Configuration for Bot Filtering.
+type botFilter struct {
+	Allow               []string
+	Deny                []string
+	Description         string
+	AllowEmptyUserAgent bool
+	UseDefaultDeny      bool
+}
+
+func (b *botFilter) toProtoConfig() *pb.MiddlewareConfiguration_BotFilter {
+	if b == nil {
+		return nil
+	}
+	return &pb.MiddlewareConfiguration_BotFilter{
+		Allow:               b.Allow,
+		Deny:                b.Deny,
+		Description:         b.Description,
+		AllowEmptyUserAgent: b.AllowEmptyUserAgent,
+		UseDefaultDeny:      b.UseDefaultDeny,
+	}
+}
+
+// BotFilter configures botfilter for this edge.
```

Comment:
Idiomatically, `BotFilter -> WithBotFilter`.

This should also provide explanation of what each of the arguments do since it's not obvious.

We could probably link to public module documentation if/when that's available instead of directly documenting it here if that seems easier.